src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -1282,6 +1308,59 @@ bool CommandLineInterface::GenerateOutput(
   return true;
 }
 
+bool CommandLineInterface::GenerateDependencyManifestFile(
+    const vector<const FileDescriptor*> parsed_files,
+    DiskSourceTree * source_tree) {
+  FileDescriptorSet file_set;
+
+  set<const FileDescriptor*> already_seen;
+  for (int i = 0; i < parsed_files.size(); i++) {
+    GetTransitiveDependencies(parsed_files[i],
+                              false,
+                              &already_seen, file_set.mutable_file());
+  }
+
+  int fd;
+  do {
+    fd = open(dependency_manifest_name_.c_str(),
+              O_WRONLY | O_CREAT | O_TRUNC | O_BINARY, 0666);
+  } while (fd < 0 && errno == EINTR);
+
+  if (fd < 0) {
+    perror(dependency_manifest_name_.c_str());
+    return false;
+  }
+
+  io::FileOutputStream out(fd);
+  io::Printer printer(&out, '$');
+
+  string output_filename = dependency_manifest_name_;
+  if (output_filename.compare(0, 1, "/") != 0) {
+    // Convert relative path to absolute path before print.
+    printer.Print("$working_directory$/$output_filename$:",
+                  "working_directory", get_current_dir_name(),
+                  "output_filename",output_filename);
```

Comment:
The current format is
^output_file_full_path: input1_full_path\\\n input2_full_path$
Which part is not good?